topiary/commandline_args/outputs.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Common commandline arguments for output files
+"""
+
+from __future__ import print_function, division, absolute_import
+
+
+def add_output_args(arg_parser):
+    output_group = arg_parser.add_argument_group(
+        title="Output",
+        description="How and where to write results")
+
+    output_group.add_argument(
+        "--output-csv",
+        default=None,
+        help="Path to output CSV file")
+
+    output_group.add_argument(
+        "--output-html",
+        default=None,
+        help="Path to output HTML file")
+
+    output_group.add_argument(
+        "--output-csv-sep",
+        default=",",
+        help="Separator for CSV file")
+
+    output_group.add_argument(
+        "--subset-output-columns",
+        nargs="*")
+
+    output_group.add_argument(
+        "--rename-output-column",
+        nargs=2,
+        action="append",
+        help=(
+            "Rename original column (first parameter) to new"
+            " name (second parameter)"))
+
+    output_group.add_argument(
+        "--print-columns",
+        default=False,
+        action="store_true",
+        help="Print columns before writing data to file(s)")
+
+    return output_group
+
+def write_outputs(
+        df,
+        args,
+        print_df_before_filtering=False,
+        print_df_after_filtering=False):
+    if print_df_before_filtering:
+        print(df)
+
+    if args.subset_output_columns:
+        for column in args.subset_output_columns:
+            if column not in df.columns:
```

Comment:
Good point...but currently the columns in the dataframe depend on which arguments get passed (e.g. is there a gene expression FPKM? a transcript FPKM?). It might be easier to *always* have the same column but leave some empty if their necessary data is missing. What do you think?

^ though immediately it occurs to me that `isovar` adds a lot of columns to the DataFrame, and seeing those if you don't want to use isovar might be messy.